ArduCopter/events.cpp

```diff
@@ -306,6 +306,77 @@ void Copter::gpsglitch_check()
     }
 }
 
+// executes lean-angle failsafe if lean angle is too high for longer than a set threshold
+// indicates the wind is too high to fly safely in planck mode
+void Copter::failsafe_lean_check(uint16_t duration_ms)
+{
+    // trigger with X seconds of failures while in a PLANCK mode
+    bool valid_mode = (control_mode == Mode::Number::PLANCKTRACK ||
+                       control_mode == Mode::Number::PLANCKRTB ||
+                       control_mode == Mode::Number::PLANCKLAND);
+
+    if (attitude_control->lean_angle()*100 > attitude_control->planck_lean_angle_max()){
+        if(failsafe.lean_high_time_ms<=0)
+        {
+            failsafe.lean_high_time_ms = AP_HAL::millis();
+        }
+    }
+    else{
+        failsafe.lean_high_time_ms = 0;
+    }
+
+    bool timeout = uint16_t(AP_HAL::millis()-failsafe.lean_high_time_ms) > g.planck_angle_fs_to;
```

Comment:
Looks like there are some signed/unsigned 16/32 bit differences here.  Be really careful

Reply:
ah, that's a leftover fix when I was using 16. It should all be 32 now, as I was seeing the ~64 second repeat of the failsafe. I'll remove the cast